src/Tool/Program.cs

```diff
@@ -35,6 +36,28 @@ public static bool TelemetryOptOut
             => true;
 #endif
 
+        // Creates dictionary of kernelspec file names to the embedded resource path
+        // of all embedded resources found in `Kernel.csproj`.
+        public static Dictionary<string, string> GetEmbeddedKernelResources()
+        {
+            var res = new Dictionary<string, string>();
+            foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
+            {
+                if (String.Equals(assembly.GetName().Name, "Microsoft.Quantum.IQSharp.Kernel"))
+                {
+                    foreach (var resName in assembly.GetManifestResourceNames())
+                    {
+                        // Compute "filename.extension"
+                        // Note: WithKernelSpecResources cannot handle keys that include directories
+                        // so we only use the filename without directory path.
+                        var fileName = String.Join(".", resName.Split(".").TakeLast(2));
+                        res.Add(fileName, resName);
+                    }
+                }
+            }
+            return res;
+        }
```

Comment:
If you want to be clever with Linq, I think you could do this all in one line. Something like:
```
        public static Dictionary<string, string> GetEmbeddedKernelResources() =>
            AppDomain.CurrentDomain.GetAssemblies()
                .Single(asm => asm.GetName().Name == "Microsoft.Quantum.IQSharp.Kernel")
                .GetManifestResourceNames()
                .ToDictionary(resName => string.Join(".", resName.Split(".").TakeLast(2)));
```

Reply:
@rmshaffer wow, that's is really nice haha! Thanks for the suggestion!

Reply:
Linq allows you to do ridiculously complicated things with very compact syntax. Highly recommend. 👍